hyperspy/models/edsmodel.py

```diff
@@ -0,0 +1,881 @@
+
+        if calibrate == 'energy':
+            bound = (bound / eV2keV) * self.units_factor
+            free = self.free_xray_lines_energy
+            fix = self.fix_xray_lines_energy
+        elif calibrate == 'sub_weight':
+            free = self.free_sub_xray_lines_weight
+            fix = self.fix_sub_xray_lines_weight
+        elif calibrate == 'width':
+            bound = (bound / eV2keV) * self.units_factor
+            free = self.free_xray_lines_width
+            fix = self.fix_xray_lines_width
+
+        free(xray_lines=xray_lines, bound=bound)
+        if kind == 'single':
+            self.fit(bounded=True, fitter='mpfit', **kwargs)
+        elif kind == 'multi':
+            self.multifit(bounded=True, fitter='mpfit', **kwargs)
+        fix(xray_lines=xray_lines)
+
+    def get_lines_intensity(self,
+                            xray_lines=None,
+                            plot_result=False,
+                            **kwargs):
+        """
+        Return the fitted intensity of the X-ray lines.
+
+        Return the area under the gaussian corresping to the X-ray lines
+
+        Parameters
+        ----------
+        xray_lines: list of str or None or 'from_metadata'
+            If None, all main X-ray lines (alpha)
+            If 'from_metadata', take the Xray_lines stored in the `metadata`
+            of the spectrum. Alternatively, provide an iterable containing
+            a list of valid X-ray lines symbols.
+        plot_result : bool
+            If True, plot the calculated line intensities. If the current
+            object is a single spectrum it prints the result instead.
+        kwargs
+            The extra keyword arguments for plotting. See
+            `utils.plot.plot_signals`
+
+        Returns
+        -------
+        intensities : list
+            A list containing the intensities as Signal subclasses.
+
+        Examples
+        --------
+        >>> m.multifit()
+        >>> m.get_lines_intensity(["C_Ka", "Ta_Ma"])
+        """
+        from hyperspy import utils
+        intensities = []
+        if xray_lines is None:
+            xray_lines = []
+            for component in self.xray_lines:
+                xray_lines.append(component.name)
+        else:
+            if xray_lines == 'from_metadata':
+                xray_lines = self.spectrum.metadata.Sample.xray_lines
+            xray_lines = filter(lambda x: x in [a.name for a in
+                                self], xray_lines)
+        if xray_lines == []:
+            raise ValueError("These X-ray lines are not part of the model.")
+        for xray_line in xray_lines:
+            element, line = utils_eds._get_element_and_line(xray_line)
+            line_energy = self.spectrum._get_line_energy(xray_line)
+            data_res = self[xray_line].A.map['values']
+            if self.axes_manager.navigation_dimension == 0:
+                data_res = data_res[0]
+            img = self.spectrum.isig[0:1].integrate1D(-1)
```

Comment:
...While that is true, I feel we unnecessary integrate and the purpose of the code is not obvious from the first glance. Instead I would write 
```python
img = self._get_navigation_signal(data_res)
img.axes_manager.set_signal_dimension(0)
```